uber/site_sections/preregistration.py

```diff
@@ -378,7 +378,14 @@ def pay_for_extra_members(self, session, payment_id, stripeToken):
             raise HTTPRedirect('group_members?id={}&message={}', group.id, 'You payment has been accepted and the badges have been added to your group')
 
     def transfer_badge(self, session, message='', **params):
-        old = session.attendee(params['id'])
+        try:
+            if not params.get('id'):
+                raise NoResultFound
+            old = session.attendee(params['id'])
+        except (NoResultFound, ValueError):
+            log.debug('transfer_badge received invalid params/id: %s' % repr(params))
+            raise HTTPRedirect('confirmation_not_found?id={}', params.get('id', 'unknown'))
```

Comment:
I would use 'none' here instead of 'unknown', since if there's no `id` in params it means they accessed the page without an ID attached to the URL.